src/lib/px4_eigen.h

```diff
@@ -34,18 +34,23 @@
 /**
  * @file px4_eigen.h
  *
- * Compatability header to make Eigen compile on the PX4 stack
+ * @brief Compatibility header to make Eigen compile on the PX4 stack
  * @author Johan Jansen <jnsn.johan@gmail.com>
- */
+ * @author Nuno Marques <n.marques21@hotmail.com>
+ **/
 
 #pragma once
 
 #pragma GCC diagnostic push
+#ifndef RAND_MAX
 #define RAND_MAX __RAND_MAX
+#endif
 #pragma GCC diagnostic ignored "-Wshadow"
 #pragma GCC diagnostic ignored "-Wfloat-equal"
+#ifndef _GLIBCXX_USE_C99_FP_MACROS_DYNAMIC
 #define _GLIBCXX_USE_C99_FP_MACROS_DYNAMIC 1
+#endif
 
-#include <eigen/Eigen/Core>
-#include <eigen/Eigen/Geometry>
+#include <eigen/Eigen/Eigen>
+#include <eigen/unsupported/Eigen/CXX11/Tensor>
```

Comment:
Just to complement this:
I send this to Eigen mailing list:
> I want to ask the dev team what's the schedule and timeline to push Tensors code to the main code of Eigen based on the parameters I present on the issue. What's in stake is their usage on critical and fail-safe system projects, as avionics, so it's important to understand the current state of development, what's in stake and what is needed so they can be used on this kind of systems.

Here's the response of one of the developers of Tensors:

> Sorry for taking so long to answer. Although I can't give you specific details, we rely extensively on tensors in several applications, and it works quite well for us :) That said, the code is provided 'as is' and comes without warranty.